src/Aspire.Hosting/Dashboard/DashboardAuthenticationOptionsValidator.cs

```diff
@@ -0,0 +1,33 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace Aspire.Hosting.Dashboard;
+
+internal class DashboardAuthenticationOptionsValidator(IConfiguration configuration) : IValidateOptions<DashboardAuthenticationOptions>
+{
+    public ValidateOptionsResult Validate(string? name, DashboardAuthenticationOptions options)
+    {
+        if (configuration[KnownEnvironmentVariables.AspNetCoreUrls] is not { } applicationUrls)
+        {
+            throw new DistributedApplicationException($"AppHost does not have applicationUrl in launch profile, or {KnownEnvironmentVariables.AspNetCoreUrls} environment variable set.");
+        }
+
+        var firstApplicationUrl = applicationUrls.Split(";").First();
+
+        if (!Uri.TryCreate(firstApplicationUrl, UriKind.Absolute, out var parsedFirstApplicationUrl))
+        {
+            throw new DistributedApplicationException($"AppHost applicationUrl in launch profile, or {KnownEnvironmentVariables.AspNetCoreUrls} environment variable could not be parsed as a URI.");
```

Comment:
Consider:

> Unable to parse launch URI from either applicationUrl in the launch profile, or the ASPNETCORE_URLS environment variable.

or just add another comma:

> AppHost applicationUrl in launch profile, or ASPNETCORE_URLS environment variable, could not be parsed as a URI.

Consider also showing the URI in the message, unless someone feels that's leaking private info.